### PR TITLE
Use webpack-bundle-analyzer for production builds

### PR DESCRIPTION
**Pull request by leeoniya**
cause it's awesome: https://github.com/grafana/grafana/issues/32328

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -192,7 +192,7 @@
     "tslib": "2.0.3",
     "typescript": "4.1.2",
     "webpack": "4.41.5",
-    "webpack-bundle-analyzer": "3.6.0",
+    "webpack-bundle-analyzer": "^4.4.0",
```
Comment (hugohaggmark) [inline package.json:195]: nit: use exact versions

File: scripts/webpack/webpack.prod.js [hunk 2]
```diff
@@ -117,6 +118,7 @@ module.exports = merge(common, {
       excludeChunks: ['manifest', 'dark', 'light'],
       chunksSortMode: 'none',
     }),
+    new BundleAnalyzerPlugin(),
```
Comment (dprokop) [inline scripts/webpack/webpack.prod.js:121]: Not sure how much time this adds to the build time, but maybe we should have control over this i.e. via env variable? Bundle analyzer data will be useful for us especially to detect regression between releases. Other than that the analysis can be performed on demand. WDYT?

**Review by dprokop — APPROVED** [review]
Looks good, just a nit regarding whether or not we want to run the analysis on every build.

File: scripts/webpack/webpack.prod.js (hunk 2 above)
Comment (ryantxu) [reply scripts/webpack/webpack.prod.js:121]: If this produces something we can easily track using:
https://github.com/grafana/grafana/blob/master/scripts/ci-frontend-metrics.sh#L14

then I think it is OK to add by default.  If it is just the visualization, then we should add a flag for it and maybe an entry in the `package.json` scripts

**Comment by torkelo** [comment]
@leeoniya is this something that we can merge?

**Comment by leeoniya** [comment]
i think we need to stick this behind a CLI flag so it doesnt run unconditionally and bust CI builds. any suggestions?

**Pull request closed without merging**